lib/http2/connection.c

```diff
@@ -808,7 +811,10 @@ static int handle_headers_frame(h2o_http2_conn_t *conn, h2o_http2_frame_t *frame
     /* request is not complete, store in buffer */
     conn->_read_expect = expect_continuation_of_headers;
     h2o_buffer_init(&conn->_headers_unparsed, &h2o_socket_buffer_prototype);
-    h2o_buffer_reserve(&conn->_headers_unparsed, payload.headers_len);
+    if ((h2o_buffer_try_reserve(&conn->_headers_unparsed, payload.headers_len)).base == NULL) {
```

Comment:
Same as above. IIRC the upper limit here is 16KB (maximum HTTP/2 frame size that we accept).

Reply:
Confirmed. `H2O_HTTP2_SETTINGS_HOST_MAX_FRAME_SIZE` is enforced when decoding the frame in `h2o_http2_decode_frame()`. Violation of this results in `H2O_HTTP2_ERROR_FRAME_SIZE`. Thanks for the exercise!